### PR TITLE
feat(curriculum): job application form

### LINKED ISSUE #617: chore: add job application form lab to main repo
Once the prototype is merged into the curriculum expansion repo, then we can add this lab with tests to the main repo.

https://github.com/freeCodeCamp/CurriculumExpansion/pull/612/files

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes https://github.com/freeCodeCamp/CurriculumExpansion/issues/617

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+assert.isNotEmpty(document.querySelectorAll("div.container > form textarea#message"));
+```
+
+You should have a `button` element with the type `submit` for submitting the form.
+
+```js
+assert.isNotEmpty(document.querySelectorAll("div.container > form button[type='submit']"));
+```
+
+Add a `:focus` pseudo-class to the `input` and `textarea` elements to change their border color when focused. Use a list selector in the given order.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:92]:
```suggestion
You should add a `:focus` pseudo-class to the `input` and `textarea` elements to change their border color when focused. Use a list selector in the given order.
```

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+assert.equal(new __helpers.CSSHelp(document).getStyle('input:valid, select:valid, textarea:valid').getPropertyValue('border-color'), 'green')
+```
+
+The `button` element should have a `:hover` pseudo-class that changes the background color when hovered over.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle('button:hover').getPropertyValue('background-color'))
+```
+
+Use the `:checked` pseudo-class on `.radio-group input[type="radio"]` to add a border color, background color and a box shadow when the radio button is selected.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:116]:
```suggestion
You should use the `:checked` pseudo-class on `.radio-group input[type="radio"]` to add a border color, background color and a box shadow when the radio button is selected.
```

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+```js
+const style = new __helpers.CSSHelp(document).getStyle('.radio-group input[type="radio"]:checked');
+assert.isNotEmpty(style.getPropertyValue('border-color'));
+assert.isNotEmpty(style.getPropertyValue('background-color'));
+assert.isNotEmpty(style.getPropertyValue('box-shadow'));
+```
+
+Use the `:checked` pseudo-class on radio buttons to change the text color of the associated `label` when the option is selected.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:125]:
```suggestion
You should use the `:checked` pseudo-class on radio buttons to change the text color of the associated `label` when the option is selected.
```

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+1. The `input`, `select` and `textarea` elements should have an `:invalid` pseudo-class that changes the border color to red when invalid input is detected.
+1. The `input`, `select` and `textarea` elements should have a `:valid` pseudo-class that changes the border color to green when valid input is entered.
+1. The `button` element should have a `:hover` pseudo-class that changes the background color when hovered over.
+1. Use the `:checked` pseudo-class on `.radio-group input[type="radio"]` to add a border color, background color and a box shadow when the radio button is selected.
+1. Use the `:checked` pseudo-class on radio buttons to change the text color of the associated `label` when the option is selected.
+1. The `button` element should have a `:disabled` pseudo-class that changes the background color when the form is incomplete or invalid.
+1. Add an `:nth-child` pseudo-class to the `input` elements to style the first input fields differently. (e.g., rounded corners).
+1. Add an `:nth-child` pseudo-class to the `textarea` element to style the fifth textarea field differently. (e.g., border-bottom-left-radius).
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:32]: I would completely remove this user story and related test as only one `textarea` has been required so far.

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+1. You should have a `textarea` element with the id `message` for entering a message.
+1. You should have a `button` element with the type `submit` for submitting the form.
+1. Add a `:focus` pseudo-class to the `input` and `textarea` elements to change their border color when focused.
+1. The `input`, `select` and `textarea` elements should have an `:invalid` pseudo-class that changes the border color to red when invalid input is detected.
+1. The `input`, `select` and `textarea` elements should have a `:valid` pseudo-class that changes the border color to green when valid input is entered.
+1. The `button` element should have a `:hover` pseudo-class that changes the background color when hovered over.
+1. Use the `:checked` pseudo-class on `.radio-group input[type="radio"]` to add a border color, background color and a box shadow when the radio button is selected.
+1. Use the `:checked` pseudo-class on radio buttons to change the text color of the associated `label` when the option is selected.
+1. The `button` element should have a `:disabled` pseudo-class that changes the background color when the form is incomplete or invalid.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:30]: Does it work without using JS?

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:30]: I don't think it does

Comment (a2937) [reply curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md]: What parts need JS? From what I see in this section right here, it is completely possible with HTML and CSS.

Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md]: `yeah, I'm removing it considering this is a html/css project

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+**User Stories:**
+
+1. You should have a `div` element with the class `container`.
+1. Inside the `div` element, you should have a `form` element.
+1. The form should contain an `input` element with the type `text` and the id `name` for entering the user's full name.
+1. You should have another `input` element with the type `email` and the id `email` for entering the user's email address.
+1. The form should include a `select` element with the id `position` that allows users to select a job position.
+1. You should have a `fieldset` element with class of `radio-group`.
+1. Inside `.radio-group` you should have a set of `input` elements with the type `radio` for selecting availability options (e.g., Full-Time, Part-Time). The group `name` sshould be `availability`.
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:21]:
```suggestion
1. Inside `.radio-group` you should have a set of `input` elements with the type `radio` and relevant labels for selecting availability options (e.g., Full-Time, Part-Time). The group `name` should be `availability`.
```

File: curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+1. You should have a `textarea` element with the id `message` for entering a message.
+1. You should have a `button` element with the type `submit` for submitting the form.
+1. Add a `:focus` pseudo-class to the `input` and `textarea` elements to change their border color when focused.
+1. The `input`, `select` and `textarea` elements should have an `:invalid` pseudo-class that changes the border color to red when invalid input is detected.
+1. The `input`, `select` and `textarea` elements should have a `:valid` pseudo-class that changes the border color to green when valid input is entered.
+1. The `button` element should have a `:hover` pseudo-class that changes the background color when hovered over.
+1. Use the `:checked` pseudo-class on `.radio-group input[type="radio"]` to add a border color, background color and a box shadow when the radio button is selected.
+1. Use the `:checked` pseudo-class on radio buttons to change the text color of the associated `label` when the option is selected.
+1. Add an `:nth-child` pseudo-class to the `input` elements to style the first input fields differently. (e.g., rounded corners).
+
```
Comment (zairahira) [inline curriculum/challenges/english/25-front-end-development/lab-job-application-form/66faac4139dbbd5dd632c860.md:31]: Add the note for consistency across the labs:

```
**Note:** Be sure to link your stylesheet in your HTML and apply your CSS.
```

**Review by vignesh1507 — COMMENTED** [review]
Thanks for making the changes, appreciate the effort @ilenia-magoni

**Comment by ilenia-magoni** [comment]
@vignesh1507 thank your for your review, but you should avoid that, you are not a maintainer for this repo and can't make the necessary approving review

**Review by a2937 — APPROVED** [review]
Everything looks good to me.

**Review by Dario-DC — APPROVED** [review]
LGTM :+1:

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**